src/main/java/org/folio/circulation/resources/TenantAPI.java

```diff
@@ -0,0 +1,54 @@
+package org.folio.circulation.resources;
+
+import static org.folio.circulation.support.http.server.JsonHttpResponse.created;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.folio.circulation.support.RouteRegistration;
+import org.folio.util.pubsub.PubSubClientUtils;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class TenantAPI extends org.folio.rest.impl.TenantAPI {
+  private static final Logger logger = LoggerFactory.getLogger(TenantAPI.class);
+
+  public void register(Router router) {
+    RouteRegistration routeRegistration = new RouteRegistration("/_/tenant", router);
+    routeRegistration.create(this::postTenant);
+  }
+
+  public void postTenant(RoutingContext routingContext) {
+    Map<String, String> headers = routingContext.request().headers().entries().stream()
+      .collect(Collectors.toMap(entry -> entry.getKey().toLowerCase(), Map.Entry::getValue));
+
+    Vertx vertx = routingContext.vertx();
+
+    vertx.executeBlocking(
+      promise -> registerModuleToPubsub(headers, vertx).setHandler(promise::complete),
+      result -> created(new JsonObject()).writeTo(routingContext.response())
```

Comment:
What will happen if registration is failed?

Reply:
It'll be ignored, which I think is incorrect. I've added a piece that checks for the result and returns server error if registration failed. This way circulation won't be registered in Okapi if it can't register in PubSub.

Reply:
Is this acceptable?

As I understand it, this would stop automated patron blocks from working. Is my understanding reasonable?

Reply:
If we're unable to register in PubSub we won't be able to publish messages and they will be lost forever (in current implementation) which can cause some patrons to be incorrectly blocked/unblocked. It's probably not the end of the world right now, but as we start using PubSub for more and more features I think this opens a broader issue of synchronization in messaging systems. One of the possible solutions I can think of is to store events in a temporary storage (e.g. in mod-circulation-storage) and delete them once they're published - this way they won't be lost. 
@marcjohnson-kint I can revert this code to make /_/tenant succeed regardless of PubSub registration and create stories for temporary event storage. What are your thoughts?

Reply:
> It's probably not the end of the world right now

Has this been agreed with the PO for the work, that automated patron blocks might silently fail to work under these circumstances?

> but as we start using PubSub for more and more features I think this opens a broader issue of synchronization in messaging systems

Totally agree, that's why I think the transient nature of the event publishing in the current design is a significant challenge.

> One of the possible solutions I can think of is to store events in a temporary storage (e.g. in mod-circulation-storage) and delete them once they're published - this way they won't be lost.

I agree that is a possibility. It is likely only part of the challenge. As an example, if the publisher could not be registered at all when the module is enabled for a tenant, then there would need to be a process for attempting that registration at a later point.

> I can revert this code to make /_/tenant succeed regardless of PubSub registration and create stories for temporary event storage. What are your thoughts?

Personally, my default in these kinds of circumstances is to fail hard and fast. I would likely choose to fail the enabling of the module unless folks were aware of this trade off and it had been agreed that this kind of silent failure is acceptable.

Reply:
@marcjohnson-kint No, it hasn't been agreed with PO. "Fail hard and fast" probably is a better choice then?

BTW, I just noticed that it wasn't actually failing together with PubSub registration. I've changed this piece and added tests to demonstrate both "fail" and "success" scenarios.

Reply:
> No, it hasn't been agreed with PO. "Fail hard and fast" probably is a better choice then?

Until FOLIO has some general guidance around this topic that would be my personal preference. There might be circumstances where that is not appropriate, I think those should be discussed with the relevant PO's (and maybe Technical Leads)

Reply:
@TarasSpashchenko also confirmed that until we come up with a more sophisticated approach, fail fast is the way to go.